test/common.js

```diff
@@ -21,6 +76,8 @@ if (process.env.TEST_THREAD_ID) {
 }
 exports.tmpDir = path.join(exports.testDir, exports.tmpDirName);
 
+refreshTmpDir();
```

Comment:
FWIW, I've got this (tmp dir refreshing handled by individual tests so it only happens for tests that use the tmp dir) ready to go once this PR goes through.